packages/docregistry/src/default.ts

```diff
@@ -616,6 +616,24 @@ export class DocumentWidget<
     if (args.name === 'dirty') {
       this._handleDirtyState();
     }
+    if (!this.context.model.dirty) {
+      if (!this.context.model.collaborative) {
+        if (!this.context.contentsModel?.writable) {
+          const readOnlyIndicator = createReadonlyLabel(this);
+          let roi = this.toolbar.insertBefore(
+            'kernelName',
+            'read-only-indicator',
+            readOnlyIndicator
+          );
+          if (!roi) {
+            this.toolbar.addItem('read-only-indicator', readOnlyIndicator);
+          }
+          Notification.warning(
+            this._trans.__(`%1 is a read-only document.`, this.title.label)
+          );
```

Comment:
I think that it would be better UX if this notification was only shown on an attempt to save with <kbd>ctrl</kbd> + <kbd>s</kbd>. See this fragment of Isabela's comment https://github.com/jupyterlab/jupyterlab/issues/7980#issuecomment-655703618 (emphasis mine):

>With toast notifications moving forward in 689, I think this could be a good opportunity to remind users a notebook is read-only **if they try to save**.

Further, currently the notification does not show up but just lands in the notification centre (makes the bell in status bar blue and requires two clicks to dismiss). As it would be responding to user action I think it should be shown outright.

Nice to have: it could also include a button for "Save as" command.

Reply:
I'm not sure how to deal with the notification on Ctrl + s without enabling the save commandID. Personally, I think having it disabled is more clear than enabling it and notifying if the user clicks. I did update to remove the notification, updated the caption that shows up when hovering over the save button and I added a dialog that pops up if the "save all" command is given.

What do you think? Is there a way to keep the save command disabled but still clickable?

Reply:
Thank you. FYI https://github.com/jupyterlab/jupyterlab/pull/14664 changes `isEnabled` for toolbar commands which will make it always enabled even if it appears as disabled in the menu. I think we can proceed with implementation as-is and see if it needs adjusting from user feedback.